app/models/waste_exemptions_engine/transient_address.rb

```diff
@@ -53,71 +40,32 @@ def self.create_from_grid_reference_data(data, address_type)
       TransientAddress.create(data)
     end
 
-    private_class_method def self.add_site_details(data, mode)
-      # Add x & y dependent on how the site was entered
-      data = update_xy_from_grid_reference(data) if mode == TransientAddress.modes[:auto]
-      data = update_xy_from_postcode(data) if mode == TransientAddress.modes[:manual]
-
-      # Add the grid reference for sites entered using an address
-      data = update_grid_reference_from_xy(data) unless mode == TransientAddress.modes[:auto]
+    private
 
-      # Add the EA administrative area
-      data = update_area_from_xy(data)
+    def update_site_details
+      update_x_and_y
+      update_grid_reference
+      update_area_from_x_and_y if area.blank?
```

Comment:
Why do we check if `area.blank?` in-line here, and we check `valid_coordinates?(x, y)` in the `update_x_and_y` method instead? Either we do the check before, or in, or after (in the service). In any case, let's just be consistent :) 
Also the naming of the method, we are not `updating`, we are just `assigning`